src/components/app.jsx

```diff
@@ -1,46 +1,64 @@
 import React from 'react';
-import CreateCards from './create-cards'
-import ReviewCards from './review-cards'
-import ViewCards from './view-cards'
-import Nav from './nav'
+import CreateCards from './create-cards';
+import ReviewCards from './review-cards';
+import ViewCards from './view-cards';
+import Nav from './nav';
 
 class App extends React.Component {
-    constructor(props){
-        super(props);
-        this.state = {
-            view: 'view-cards',
-        }
-        this.setView = this.setView.bind(this);
-        this.getView = this.getView.bind(this);
-    }
+  constructor(props) {
+    super(props);
+    this.state = {
+      view: 'view-cards',
+      cssBadge: 'view-cards',
+      cards: []
+    };
+    this.setView = this.setView.bind(this);
+    this.getView = this.getView.bind(this);
+    this.saveCards = this.saveCards.bind(this);
+    this.addCards = this.addCards.bind(this);
+  }
 
-    setView(newView){
-        this.setState({
-            view: newView        
-        })
-    }
+  setView(newView) {
+    this.setState({
+      view: newView,
+      cssBadge: newView
+    });
+  }
 
-    getView(){
-        switch(this.state.view){
-            case 'create-cards':
-                return <CreateCards />;
-            case 'review-cards':
-                return <ReviewCards />;
-            case 'view-cards':
-                return <ViewCards />;
-            default:
-                return null;
-        }
+  getView() {
+    switch (this.state.view) {
+      case 'create-cards':
+        return <CreateCards setView={this.setView} addCards={this.addCards}/>;
+      case 'review-cards':
+        return <ReviewCards />;
+      case 'view-cards':
+        return <ViewCards />;
+      default:
+        return null;
     }
+  }
+
+  saveCards() {
+    let cards = this.state.cards;
+    JSON.stringify(cards);
```

Comment:
`JSON.stringify` doesn't mutate the original variable so you need to save the returned value into a variable to save the stringified version. For example:

```JAVASCRIPT
let cards = this.state.cards;
cards = JSON.stringify(cards);
```